soc/ambiq/apollo4x/Kconfig

```diff
@@ -11,3 +11,8 @@ config SOC_SERIES_APOLLO4X
 	select CPU_HAS_ARM_MPU
 	select HAS_SWO
 	select AMBIQ_HAL
+	select CPU_HAS_DCACHE
+	select CPU_HAS_ICACHE
+	select HAS_PM
+	help
+	  Enable support for Apollo4 MCU series
```

Comment:
Can we not add things like this? This has no selection, there does not need to be help text

Reply:
Does this conflict with Zephyr documentation? If not, I think this should be left in.

Reply:
Can you quote where it says to add help text to `SOC_SERIES_*` symbols?

Reply:
If you can point to a failed compliance check or documentation that says this is not allowed, then I will update.
If not, I am considering this resolved and waiting for you to undo the requested changes.

Reply:
It will not be removed until the issue is addressed

Reply:
If you cannot point me to the documentation that says it is not allowed, then I request that you undo the requested change.

Reply:
@RichardSWheatley unless we have a precedent of another `SOC_SERIES_*` Kconfig option with a help text then consistency trumps any other criteria.

Reply:
I'm all for consistency. Also, any valid reasons other than "I think it should be this way" is fine with me.

Reply:
@carlescufi ok, so if there is a config, a variable type, then you can have a help.
Is that correct? Such as this?

![image](https://github.com/zephyrproject-rtos/zephyr/assets/12254017/7484c3ed-23fb-49a8-8020-70eff28daa78)

Reply:
You're quoting something that was out of date 7 years ago, that is remnants from when zephyr used kbuild, which was replaced with cmake in 2017, it stopped being relevant then but existing options were not updated, and now with hwmv2 it's completely irrelevant because these are hidden symbols